project-base/storefront/utils/cookies/cookiesStore.ts

```diff
@@ -1,32 +1,82 @@
 import { getCookies, setCookie } from 'cookies-next';
 import { GetServerSidePropsContext } from 'next';
 import { useEffect } from 'react';
-import { CookiesStoreState, useCookiesStore } from 'store/useCookiesStore';
+import { useCookiesStore } from 'store/useCookiesStore';
 import { v4 as uuidV4 } from 'uuid';
+import { createStore } from 'zustand/vanilla';
 
-const COOKIES_STORE_NAME = 'cookiesStore' as const;
-const THIRTY_DAYS_IN_SECONDS = 60 * 60 * 24 * 30;
+export type CookiesStoreState = {
+    lastVisitedProductsCatnums: string[] | null;
+    userIdentifier: string;
+};
+
+type CookiesStoreActions = {
+    setCookiesStoreState: (value: Partial<CookiesStoreState>) => void;
+};
+
+export type CookiesStore = CookiesStoreState & CookiesStoreActions;
 
 type CookiesType = {
     cookiesStore: string;
 };
 
+const COOKIES_STORE_NAME = 'cookiesStore' as const;
+const THIRTY_DAYS_IN_SECONDS = 60 * 60 * 24 * 30;
+
 const getDefaultInitState = (): CookiesStoreState => ({
     lastVisitedProductsCatnums: null,
     userIdentifier: uuidV4(),
 });
 
 export const getCookiesStoreState = (context?: GetServerSidePropsContext): CookiesStoreState => {
     const { cookiesStore } = getCookies(context) as CookiesType;
+    const newState = getDefaultInitState();
 
-    return cookiesStore ? JSON.parse(decodeURIComponent(cookiesStore)) : getDefaultInitState();
+    if (!cookiesStore) {
+        return newState;
+    }
+
+    return removeIncorrectCookiesStoreProperties(
+        Object.keys(newState),
+        addMissingCookiesStoreProperties(newState, JSON.parse(decodeURIComponent(cookiesStore))),
+    );
+};
+
+const addMissingCookiesStoreProperties = (
```

Comment:
I'm not sure if this function does what it's name says
It's actually only responsible for merging the previous state cookie store with the current one, so I don't see it getting the missing properties somewhere inside 🤔